DependencyInjection/Compiler/SystemInfoCollectorPass.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * File containing the SystemInfoCollectorPass class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzSupportToolsBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+class SystemInfoCollectorPass implements CompilerPassInterface
+{
+    /**
+     * Registers the SystemInfoCollector into the system info collector registry.
+     *
+     * @param \Symfony\Component\DependencyInjection\ContainerBuilder $container
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->has('support_tools.info_collector.registry')) {
+            return;
+        }
+
+        $infoCollectorRegistry = $container->findDefinition('support_tools.info_collector.registry');
+        $infoCollectors = $container->findTaggedServiceIds('support_tools.info_collector');
+
+        foreach ($infoCollectors as $id => $tags) {
+            foreach ($tags as $attributes) {
+                $infoCollectorRegistry->addMethodCall('addToRegistry', [new Reference($id), $attributes['alias']]);
```

Comment:
A performance improvement we try to apply: instead of adding N method calls, add one method call with N collectors. You could just make `addToRegistry` accept a hash of collectors, indexed by identifier.